cmd/postcli/main.go

```diff
@@ -344,19 +410,27 @@ func cmdVerifyPos(opts config.InitOpts, fraction float64, logger *zap.Logger) {
 	if n != ed25519.PrivateKeySize {
 		log.Fatalf("size of key (%d) not expected size %d\n", n, ed25519.PrivateKeySize)
 	}
-	pub := ed25519.NewKeyFromSeed(dst[:ed25519.SeedSize]).Public().(ed25519.PublicKey)
+	return ed25519.NewKeyFromSeed(dst[:ed25519.SeedSize]).Public().(ed25519.PublicKey), nil
+}
+
+func cmdVerifyPos(opts config.InitOpts, fraction float64, logger *zap.Logger) {
+	log.Println("cli: verifying", edKeyFileName)
+	pub, err := loadKey()
+	if err != nil {
+		log.Fatalf("failed to load public key from %s: %s\n", edKeyFileName, err)
+	}
```

Comment:
Now that I think of this, it could just read the key from the metadata JSON file. There is no reason to require the identity.key to exist in order to verify the POS data.

Reply:
I changed the code to skip verification of the key when it is missing; but left the check if a key is present to give a user the option to still verify the key against the initialised data (if i.e. they forgot which key belongs to which PoST).